### PR TITLE
Allow to create a new value based on a list of unions

### PR DESCRIPTION
**Pull request by HRogge**
Allow to create a new value based on a list of unions as long as they contain the same type

### CODE REVIEW

**Comment by dhubler** [comment]
is there a unit test you can create/update?

**Comment by HRogge** [comment]
There is node/value_test.go, but I am not sure how to use "meta.Builder" to build a union type...

Easiest way to test the new constructor would be to have a int/string union type and then try to assign both types and test the resulting slice.

**Comment by dhubler** [comment]
unionlist should be:

    l := b.LeafList(m, "l")
    dt := b.Type(l, "union")
    
constructor is private but even if public, wouldn't get you far because a lot of private fields.

File: node/value.go [hunk 1]
```diff
@@ -74,6 +75,24 @@ func NewValue(typ *meta.Type, v interface{}) (val.Value, error) {
 	case val.FmtUnion:
 		cvt, _, err := val.ConvOneOf(typ.UnionFormats(), v)
 		return cvt, err
+	case val.FmtUnionList:
+		if v == nil {
+			return nil, nil
+		}
+		sliceValue := reflect.ValueOf(v)
```
Comment (dhubler) [inline node/value.go]: can you move this code to toUnionList instead of making this inline to keep consistent w/other conversion handlers?

**Comment by HRogge** [comment]
I tried the following code as a test but it doesn't work
```
func TestToUnionList(t *testing.T) {
	b := &meta.Builder{}
	m := b.Module("x", nil)
	l := b.LeafList(m, "l")
	dt := b.Type(l, "union")
	if err := meta.Compile(m); err != nil {
		t.Error(err)
	}
	v, err := NewValue(dt, []int{1, 2})
	if err != nil {
		t.Error(err)
	} else if sl := v.Value().([]int); sl[0] != 1 || sl[1] != 2 {
		t.Errorf("Coersion error: %v", sl)
	}
}
```

=== RUN   TestToUnionList
    value_test.go:61: x/l - unions need at least one type
    value_test.go:65: could not coerce [1 2] into UnionList

**Comment by dhubler** [comment]
i'll take a look at this tomorrow

**Comment by dhubler** [comment]
i did look at this and unions are bit more complicated then I realized. this will take me a few more days.

**Comment by HRogge** [comment]
I think this pull-request should be closed...

**Pull request closed without merging**